sarcophagus-org/sarcophagus-v2-app
# 12 setup error boundary + sentry

# Error handling with Sentry

Adds Sentry to the app for error reporting.
[Sentry Decent org](https://sentry.io/organizations/decent-mg/issues/)

### Notes
- Added Sentry to the app
- Wrapped the app in Sentry's `ErrorBoundary`
- Added a toast message to the archaeologist list to make sure its working properly

### Testing
- Cause an error in the app
  - Go to `src/features/archaeologist/archaeologistList/index.tsx` and pass an invalid function name to a component like this 
```
      <Button
        width={20}
        size="sm"
        background="gray"
        onClick={invalidFunction}
      >
        Clear
      </Button>
```

- Expect the app to show the regular react error but also render the `ErrorFallback` component that says "Oops. There was an error". 
- Check Sentry for the error.
